src/engine/strat_engine/engine.rs

```diff
@@ -170,40 +173,46 @@ impl Engine for StratEngine {
                     .expect("We just retrieved or created a HashMap");
                 devices.insert(device, dev_node);
 
-                let rc = if let Ok((pool_name, pool)) =
-                    StratPool::setup(&dm, pool_uuid, &devices) {
-                    // We know we have a unique uuid, but the pools table requires a unique name too
-                    // so we will ensure unique before we insert as we don't want to change the
-                    // existing state if we have a conflict.
-                    //
-                    // We will also _not_ exit the daemon as we do in initialize as we were
-                    // previously up and running for some duration of time.
-                    if !self.pools.contains_name(&pool_name) {
-                        self.pools.insert(pool_name, pool_uuid, pool);
-                        Some(pool_uuid)
-                    } else {
-                        let dev_paths = devices
-                            .values()
-                            .map(|p| p.to_str().expect("Expecting valid device path!"))
-                            .collect::<Vec<&str>>()
-                            .join(", ");
 
-                        self.incomplete_pools.insert(pool_uuid, devices);
-
-                        error!("udev add: duplicate pool name {:?} for uuid {:?}, \
-                                devices[{}], failing to setup complete pool!",
-                               pool_name,
-                               pool_uuid,
-                               dev_paths);
-                        if let Err(e) = pool.teardown(&dm) {
-                            error!("Error while tearing down pool with duplicate name! {:?}!",
-                                   e);
+                let rc = match StratPool::setup(&dm, pool_uuid, &devices) {
+                    Ok((pool_name, pool)) => {
+                        // We know we have a unique uuid, but the pools table requires a unique
+                        // name too so we will ensure unique before we insert as we don't want to
+                        // change the existing state if we have a conflict.
+                        //
+                        // We will also _not_ exit the daemon as we do in initialize as we were
+                        // previously up and running for some duration of time.
+                        if !self.pools.contains_name(&pool_name) {
+                            self.pools.insert(pool_name, pool_uuid, pool);
+                            Some(pool_uuid)
+                        } else {
+                            let dev_paths = devices
+                                .values()
+                                .map(|p| p.to_str().expect("PathBuf.to_str for device path fail"))
```

Comment:
The expect() message should explain why this action must logically succeed.

Reply:
@mulkieran Please state what you would like this expect statement to be, thanks!

Reply:
I've selected the same text Andy used in a previous commit that has the same expectation of success.